tests/unit/amo/components/TestAddon.js

```diff
@@ -757,10 +758,11 @@ describe(__filename, () => {
     expect(src).toEqual('default-64.png');
   });
 
-  it('renders a theme preview as an img', () => {
+  it('renders a lightweight theme preview as an image', () => {
     const root = shallowRender({
       addon: createInternalAddon({
         ...fakeTheme,
+        previews: [],
```

Comment:
We probably need a test case for when there are `previews` no?

Reply:
well static theme uses data from previews and lightweight theme uses value from theme_data

in the current api response lightweight theme shows previews as an empty array so that's this case here and underneath I have a test case for the static theme scenario (which uses the preview array). 

will those work?

Reply:
ah I see, but then maybe the code in the component should be a bit more defensive?

Reply:
i also added the following comment to describe the purpose of this:

```javascript 
// The 'previews' field is not currently being used by lightweight themes
// So here we are just overridding the fakeAddon values to mimic the API
// response
previews: [],
...
```

Reply:
>ah I see, but then maybe the code in the component should be a bit more defensive?

I guess my thinking was instead of checking for each type, and using that to determine the previewURL, I would make it a little bit more flexible for when the migration happens (moving lightweight themes to static themes, where they WILL have this field in the near future) ... if that makes sense...

Reply:
it does I guess, but it leaves code paths untested.

Reply:
sorry to clarify.. it actually does check type of (lightweight) theme as a fallback (which means migration hasn't happened yet). 

I can add this fallback scenario as an additional test..